Google.Api.Generator/Utils/ResourcePattern.cs

```diff
@@ -131,7 +139,19 @@ void Check(bool cond, string msg)
                 string.Join("", ParameterNamesWithSuffix.Zip(Separators, (p, s) => $"{{{p}}}{s}").Append($"{{{ParameterNamesWithSuffix[^1]}}}"));
         }
 
-        public ResourcePattern(string pattern) => Segments = pattern.Split('/').Select(Segment.Create).ToImmutableList();
+        public ResourcePattern(string pattern)
+        {
+            Segments = pattern.Split('/').Select(Segment.Create).ToImmutableList();
+            if (Segments.OfType<ResourceIdSegment>().Any())
+            {
+                // Perform tight (standard) validation if there are any resource-id path-segments in this pattern.
+                // Loose validation is required to support pubsub's legacy use of the pattern '_deleted-topic_'
+                foreach (var segment in Segments.OfType<CollectionIdentifierSegment>())
+                {
+                    segment.Validate(tightValidation: true);
+                }
```

Comment:
```suggestion
                foreach (var segment in Segments)
                {
                    segment.Validate(tightValidation: segment is CollectionIdentifierSegment);
                }
```

Reply:
This isn't doing quite the same thing - only the `CollectionIdentiferSegment` type has a `Validate(...)` method, so the `OfType<CollectionIdentiferSegment>` in the `foreach` is required to make `segment` the correct type.